hamilflow/models/brownian_motion.py

```diff
@@ -183,17 +185,27 @@ def _trajectory(self, n_new_steps: int, seed: int) -> np.ndarray:
 
         return trajectory
 
-    def __call__(self, n_steps: int, seed: int = 42) -> pd.DataFrame:
-        """Simulate the coordinates of the particle
+    def generate_from(self, n_steps: int, seed: int = 42) -> pd.DataFrame:
+        """generate data from a set of interpretable params for this model
 
         :param n_steps: total number of steps to be simulated, including the inital step.
         :param seed: random generator seed for the stochastic process.
             Use it to reproduce results.
         """
-        trajectory = self._trajectory(n_new_steps=n_steps - 1, seed=seed)
+        time_steps = np.arange(0, n_steps) * self.system.delta_t
+
+        return self(t=time_steps, seed=seed)
+
+    def __call__(self, t: TypeTime, seed: int = 42) -> pd.DataFrame:
+        """Simulate the coordinates of the particle
 
-        df = pd.DataFrame(trajectory, columns=self._axis_names)
+        :param t: the time sequence to be used to generate data, 1-D array like
+        :param seed: random generator seed for the stochastic process.
+            Use it to reproduce results.
+        """
+        n_steps = np.array(t).size
+        trajectory = self._trajectory(n_new_steps=n_steps - 1, seed=seed)
 
-        df["t"] = np.arange(0, n_steps) * self.system.delta_t
+        df = pd.DataFrame(trajectory, columns=self._axis_names).assign(t=t)
```

Comment:
This `assign` will copy the `DataFrame` that you just created. To be more memory economic, we can use https://numpy.org/doc/stable/reference/generated/numpy.concatenate.html to append the `t` array to `trajectory` and create the `DataFrame` once for all.

Reply:
I think concat reduces the readability, so I reverted to my previous code for memory efficiency in 5fb33a4